miso-web/src/main/webapp/scripts/hot_order.js

```diff
@@ -0,0 +1,101 @@
+        readOnly: true,
+        include: true,
+        unpack: function(order, flat, setCellMeta) {
+          flat.poolName = order.pool.name;
+        },
+        pack: function(order, flat, errorHandler) {
+        }
+      }, {
+        header: 'Pool Alias',
+        data: 'poolAlias',
+        readOnly: true,
+        include: true,
+        unpack: function(order, flat, setCellMeta) {
+          flat.poolAlias = order.pool.alias;
+        },
+        pack: function(order, flat, errorHandler) {
+        }
+      }, {
+        header: 'Platform',
+        data: 'platform',
+        type: 'dropdown',
+        trimDropdown: false,
+        validator: HotUtils.validator.requiredAutocomplete,
+        source: [''],
+        include: true,
+        unpack: function(order, flat, setCellMeta) {
+        },
+        pack: function(order, flat, errorHandler) {
+        },
+        depends: '*start',
+        update: function(order, flat, flatProperty, value, setReadOnly, setOptions, setData) {
+          var platforms = Constants.platforms.filter(function(platform) {
+            return platform.platformType == order.pool.platformType && platform.active;
+          }).map(function(platform) {
+            return platform.instrumentModel;
+          }).sort();
+          setOptions({
+            source: platforms
+          });
+        }
+      }, {
+        header: 'Sequencing Parameters',
+        data: 'sequencingParameters',
+        type: 'dropdown',
+        trimDropdown: false,
+        validator: HotUtils.validator.requiredAutocomplete,
+        source: [''],
+        include: true,
+        unpack: function(order, flat, setCellMeta) {
+          if (order.parameters) {
+            flat.sequencingParameters = Utils.array.findFirstOrNull(function(param) {
+              return param.id == order.parameters.id;
+            }, Constants.sequencingParameters);
+          }
+        },
+        pack: function(order, flat, errorHandler) {
+          order.parameters = Utils.array.findFirstOrNull(function(param) {
+            return param.name == flat.sequencingParameters;
+          }, Constants.sequencingParameters);
+        },
+        depends: 'platform',
+        update: function(order, flat, flatProperty, value, setReadOnly, setOptions, setData) {
+          if (value) {
+            var platform = Utils.array.findFirstOrNull(function(platform) {
+              return platform.instrumentModel == value;
+            }, Constants.platforms);
+            var params = Constants.sequencingParameters.filter(function(parameters) {
+              return parameters.platform.id == platform.id;
+            }).map(function(parameters) {
+              return parameters.name;
+            }).sort();
+          } else {
+            var params = [];
+          }
+          setOptions({
+            source: params
+          });
+        }
+      }, HotUtils.makeColumnForInt('Partitions', true, 'partitions', HotUtils.validator.requiredNumber)];
```

Comment:
Hmm, we've been able to enter negative numbers on all our `requiredNumber`s this whole time... so looks like it's not an issue in production, but might be nice to guard against negative numbers in the validator.

Reply:
`requiredNumber` allowed all kinds of numbers, but in all cases where we used it, we actually wanted positive integers only. I changed it to `requiredPositiveInt`